lib/dependencies/HarmonyTopLevelThisParserPlugin.js

```diff
@@ -0,0 +1,38 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+"use strict";
+
+const walk = require("acorn/dist/walk");
+const ConstDependency = require("./ConstDependency");
+
+const rewriteThisVisitor = {
```

Comment:
Instead of a custom walker in the plugin, add the ability to parse top-level-this to the Parser with a new hook i. e. `topLevelThis`.